UI/cypress/support/commands.js

```diff
@@ -24,25 +24,18 @@
 // -- This will overwrite an existing command --
 // Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
 
-import { removeNewLine } from "./utils";
-
-import { getJumpToLocationId } from "./utils";
+import { removeNewLine, getJumpToLocationId, applyBasicAuth } from "./utils";
 
 Cypress.Commands.overwrite("visit", (originalFn, url, options) => {
-  const username = Cypress.env("username");
-  const password = Cypress.env("password");
-
-  if (username && password) {
-    options = options || {};
-    options.auth = {
-      username: username,
-      password: password,
-    };
-  }
-
+  options = applyBasicAuth(options);
   return originalFn(url, options);
 });
 
+Cypress.Commands.add("requestWithBasicAuth", (options) => {
```

Comment:
Could we not just use `Cypress.Commands.overwrite("request"...` much like we do with visit? That would make it work with any `request` in the future.

Reply:
I tried that [approach](https://github.com/DFE-Digital/find-a-tuition-partner/pull/327/commits/cb9f7444385014e0b8b8fd4d762a25bb13618301); however, It didn't work. Looking at the [logs](https://github.com/DFE-Digital/find-a-tuition-partner/actions/runs/4813444926/jobs/8570013729), some of our end-to-end tests make a call to the external URL, and with our basic auth in the header in the request they failed gives us 403 error.

Reply:
I see what you mean - `request` has a much more complicated signature which would need to be replicated, and is also used to access external resources. Maybe try something like this?
```js
Cypress.Commands.overwrite("request", (originalFn, ...args) => {
  if(typeof args[0] == 'object') {
    let options = {...args[0]};

    const url = new URL(options.url, Cypress.config().baseUrl);

    if(url.toString().startsWith(Cypress.config().baseUrl)) {
      options = applyBasicAuth(options);
    }

    return originalFn(options);
  }

  return originalFn.apply(null, args);
});
```